main.py

```diff
@@ -0,0 +1,202 @@
+import math
+import os
+import csv
+import statistics
+
+
+def main():
+    months = [" ", "Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"]
+    time = input("Enter the time you want to get files: ")
+    if "/" in time:
+        year = time.split("/")[0]
+        month = time.split("/")[1:]
+        month = int(month[0])
+        month_name = months[month]
+        monthly_files(year, month_name)
+    else:
+        year = time
+        month_name = "."
+        file_processing(month_name, year)
+
+
+def monthly_files(year, month_name):
+    files_path = r"/home/hamza/Downloads/weatherfiles/weatherfiles/"
+    filepaths = [os.path.join(files_path, name) for name in os.listdir(files_path)]
+    for path in filepaths:
+        if month_name in path:
```

Comment:
Could we have any better way of doing this? Why don't we generate filenames on basis of input and just consume those?